### PR TITLE
feat: [FFM-10128]: case sensitive tag validation

### PR DESCRIPTION
**Pull request by crystal-fong**
This PR addresses:
- Ensuring a tag name does not exceed 100 characters
- Does not contain commas in its name
- Fixes the bug where the tag identifier and name is passed in the wrong order in the patch function param
- Restricts the size of the tag dropdown so it doesn't skew the modal it's in
- Allows user to type in a space for the tag name, but the tag identifier will replace the space with underscores 

- These validation checks exists within the Flag creation wizard and within the Edit feature flag details modal

https://github.com/harness/harness-core-ui/assets/95680496/9d18b7f7-fbd9-4164-82c8-27525ec8d63d

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by crystal-fong** [comment]
trigger messagecheck

**Review by laura-mannus — APPROVED** [review]
you are getting so good at these tests!

File: src/modules/75-cf/components/FlagActivation/hooks/__tests__/useEditFlagDetailsModal.test.tsx [hunk 1]
```diff
@@ -157,11 +156,104 @@ describe('useEditFlagDetailsModal', () => {
...
+      },
+      isTaggingFFOn
+    )
+
+    await userEvent.click(screen.getByRole('button', { name: openEditFlagModalBtn }))
+
+    const tagDropdown = document.querySelector('[class="bp3-input-ghost bp3-multi-select-tag-input-input"]')!
+
+    const reallyLongTagName =
+      'erthyretyyttmbtepytisqfbhyecpkuiegaadwqrebspzrfswxskupfawwylzxjbpchjcfbawbyqosuqhiafvvuqncgsnxvvovnynuzfmfrghwzwvifz'
```
Comment (Kira-Jennings) [inline src/modules/75-cf/components/FlagActivation/hooks/__tests__/useEditFlagDetailsModal.test.tsx:186]: Is this the name of that place in Wales? 😂

File: src/modules/75-cf/components/FlagActivation/hooks/__tests__/useEditFlagDetailsModal.test.tsx [hunk 2]
```diff
@@ -157,11 +156,104 @@ describe('useEditFlagDetailsModal', () => {
...
+
+    await userEvent.type(tagDropdown, reallyLongTagName)
+
+    await userEvent.click(screen.getByRole('button', { name: reallyLongTagName }))
+
+    expect(await screen.findByText('cf.featureFlags.tagging.inputErrorMessage')).toBeInTheDocument()
```
Comment (Kira-Jennings) [inline src/modules/75-cf/components/FlagActivation/hooks/__tests__/useEditFlagDetailsModal.test.tsx:192]: Is this `await` needed? Usually the wait on the userEvent above is enough

File: src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.module.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.dropdown {
+  :global {
```
Comment (Kira-Jennings) [inline src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.module.scss:2]: Does this affect all dropdowns that are rendered?

**Review by Kira-Jennings — DISMISSED** [review]

File: src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.module.scss (hunk 3 above)
Comment (crystal-fong) [reply src/modules/75-cf/components/FlagActivation/hooks/useEditFlagDetailsModal.module.scss:2]: since it's within a css class it shouldn't

**Review by laura-mannus — APPROVED** [review]

**Review by Kira-Jennings — APPROVED** [review]

**Pull request merged**